server/api/core/core.js

```diff
@@ -364,29 +363,93 @@ export default {
     return null;
   },
 
+  /**
+   * @name connectionDataStore
+   * @summary a construct to store per-connection data. Once a connection is
+   * closed, the data is lost. This is server-side companion to Sessions in the
+   * client. connectionDataStore has two methods: get and set
+   * @memberof Core
+   */
+  connectionDataStore: (function () {
+    const inMemoryCache = {};
+    const connectionKey = "connection-data";
+
+    function connectionData() {
```

Comment:
implementation reflects that of [`Meteor.userId()`](https://github.com/meteor/meteor/blob/f551a56f93246b0a8251426d8f8eb9964cbdd865/packages/accounts-base/accounts_server.js#L69-L80), with a fallback for using it outside of a connection (i.e., on startup)